eng/common/TestResources/Remove-TestResources.ps1

```diff
@@ -125,9 +158,12 @@ if (![string]::IsNullOrWhiteSpace($ServiceDirectory)) {
     if (Test-Path $preRemovalScript) {
         Log "Invoking pre resource removal script '$preRemovalScript'"
 
-        if (!$PSCmdlet.ParameterSetName.StartsWith('ResourceGroup')) {
-            $PSBoundParameters.Add('ResourceGroupName', $ResourceGroupName);
-        }
+        # Ensure that expected parameters are added
+        $null = $PSBoundParameters.TryAdd('ResourceGroupName', $ResourceGroupName);
+        $null = $PSBoundParameters.TryAdd('SubscriptionId', $SubscriptionId);
+        $null = $PSBoundParameters.TryAdd('TenantId', $TenantId);
+        $null = $PSBoundParameters.TryAdd('ProvisionerApplicationId', $ProvisionerApplicationId);
+        $null = $PSBoundParameters.TryAdd('ProvisionerApplicationSecret', $ProvisionerApplicationSecret);
```

Comment:
This is an example where the `@PSBoundParameters` needed some values that are no longer parameters